### PR TITLE
feat: Add pause, unpause type to kernel lifecycle

### PR DESCRIPTION
**Pull request by Yaminyam**
https://github.com/lablup/backend.ai/pull/629 isolated from pr
Added pause and unpause to kernel lifecycle types.

### CODE REVIEW

**Review by fregataa — COMMENTED** [review]
good!
how about let pause_kernel and unpause_kernel methods in kubernetes.agent module raise `NotImplementedError`?

File: src/ai/backend/agent/agent.py [hunk 1]
```diff
@@ -927,6 +928,39 @@ async def _handle_clean_event(self, ev: ContainerLifecycleEvent) -> None:
...
+    ) -> None:
+        pass
+
+    @abstractmethod
+    async def unpause_kernel(
+        self,
+        kernel_id: KernelId,
+        container_id: Optional[ContainerId],
+    ) -> None:
+        pass
```
Comment (kyujin-cho) [inline src/ai/backend/agent/agent.py:945]: Please implement these methods on Kubernetes backend also. If you don't have any knowledge about Kubernetes, it's okay to let implementations as blank and create separate issue stating we have to implement it later.

**Review by kyujin-cho — APPROVED** [review]

File: src/ai/backend/agent/agent.py (hunk 1 above)
Comment (kyujin-cho) [inline src/ai/backend/agent/agent.py:962]: Please implement these methods on Kubernetes backend also. If you don't have any knowledge about Kubernetes, it's okay to let implementations as blank and create separate issue stating we have to implement it later.

**Review by kyujin-cho — CHANGES_REQUESTED** [review]

**Review by s4shantanu — APPROVED** [review]

File: src/ai/backend/agent/docker/agent.py [hunk 2]
```diff
@@ -1501,18 +1548,35 @@ async def handle_action_die(kernel_id: KernelId, evdata: Mapping[str, Any]) -> N
...
+                                        self.docker_ptask_group.create_task(
+                                            handle_action_start(kernel_id, evdata),
+                                        )
                                     )
-                                )
-                            elif evdata["Action"] == "die":
-                                await asyncio.shield(
-                                    self.docker_ptask_group.create_task(
-                                        handle_action_die(kernel_id, evdata),
+                                    break
```
Comment (achimnol) [inline src/ai/backend/agent/docker/agent.py]: I remember that Python's match syntax does not 'fall-through' to the subsequent case blocks. Please remove these unnecessary `break` statements, unless you are intending to exit the entire subscription loop!

**Review by achimnol — CHANGES_REQUESTED** [review]

**Review by s4shantanu — COMMENTED** [review]
Looks good to me!!

**Comment by achimnol** [comment]
As #629 is superseded by #1373, I'm closing this as well.
Currently it's not practical to forcibly pause a container (i.e., excluding the container processes from the Linux scheduler) as this will make it unresponsive at all while the container still occupies the agent resources.

**Pull request closed without merging**